rp2040-hal/src/atomic_register_access.rs

```diff
@@ -0,0 +1,56 @@
+//! Provide atomic access to peripheral registers
+//!
+//! This feature is not available for all peripherals.
+//! See [section 2.1.2 of the RP2040 datasheet][section_2_1_2] for details.
+//!
+//! [section_2_1_2]: https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf#atomic-rwtype
+
+use core::ptr::write_volatile;
+
+/// Perform atomic XOR operation on register
+///
+/// See [section 2.1.2 of the RP2040 datasheet][section_2_1_2] for details.
+///
+/// [section_2_1_2]: https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf#atomic-rwtype
+///
+/// # Safety
+///
+/// In addition to the requirements of [core::ptr::write_volatile],
+/// `register` must point to a register providing atomic aliases.
+#[inline]
+pub unsafe fn write_xor(register: *mut u32, bits: u32) {
+    let alias = (register as usize + 0x1000) as *mut u32;
```

Comment:
`alias.offset(0x400)` ?

Reply:
alias.offset(0x400).write_volatile(bits)?

Reply:
It may be useful to allow for 8 or 16 bit writes, yes. But for now, we don't need it. As I changed the methods to `pub(crate)` visibility, they are not part of the API and therefore can be easily updated when the actual need for such writes arises.